include/systems/system.h

```diff
@@ -1927,6 +1927,22 @@ class System : public ReferenceCountedObject<System>,
    */
   void prefer_hash_table_matrix_assembly(bool preference);
 
+  /**
+   * Instructs this system to prefix solve options with its name for solvers that leverage prefixes
+   */
+  void prefix_with_name(bool value) { _prefix_with_name = value; }
+
+  /**
+   * @returns Whether we are name prefixing
+   */
+  [[nodiscard]] bool prefix_with_name() const { return _prefix_with_name; }
```

Comment:
This is the first/only place we use the `[[nodiscard]]` attribute in the library, should we add a configure test for it? I just remembered we have all this stuff
```
// Define backwards-compatible fallthrough attribute.  We could
// eventually also add support for other compiler-specific fallthrough
// attributes.
#ifdef LIBMESH_HAVE_CXX17_FALLTHROUGH_ATTRIBUTE
#define libmesh_fallthrough() [[fallthrough]]
#elif defined(LIBMESH_HAVE_DOUBLE_UNDERSCORE_ATTRIBUTE_FALLTHROUGH)
#define libmesh_fallthrough() __attribute__((fallthrough))
#else
#define libmesh_fallthrough() ((void) 0)
#endif
```
for the `[[fallthrough]]` attribute, but these attributes have been standard since C++17 so maybe time to get rid of `libmesh_fallthrough()`

Reply:
:+1: I `git grepped` for a test in the `m4` directory and didn't find one. I guess that type of grep doesn't descend into submodules.

Reply:
And it looks like we are running it.  Checking one of my `config.log` files I see the `[[nodiscard]]` function in the `failed program was:` output for a case where my `mpicxx` originally failed before being retried with `-std=gnu++17`.